.github/workflows/ant.yaml

```diff
@@ -44,19 +44,59 @@ jobs:
       - run: yarn aos:build
       - run: yarn test
 
-  notify:
+  publish:
     runs-on: ubuntu-latest
     needs: [unit, integration]
+    if: github.ref == 'refs/heads/main'
+    outputs:
+      publish-output: ${{ steps.set-publish-output.outputs.publish-output }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4.0.2
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - run: yarn --frozen-lockfile
+      - run: yarn aos:build
+        id: publish-lua-code
+      - run: yarn aos:publish
+        env:
+          ARWEAVE_PUBLISHING_KEY: ${{ secrets.ARWEAVE_PUBLISHING_KEY }}
+      - name: Capture Publish Output
+        id: set-publish-output
+        run: |
+          publish_output=$(node -e "const fs = require('fs'); const output = JSON.parse(fs.readFileSync('tools/publish-output.json', 'utf-8')); console.log(output);")
+          echo "::set-output name=publish-output::$publish_output"
+      # Add the following steps to commit and push the changelog file and create a tag
+      - name: Commit and Push Changelog
+        run: |
+          prettier --write changelogs/*.md
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add changelogs/*.md
```

Comment:
you should be able to add read/write permissions on the action to avoid any git config setup 

example here - although you shouldn't need all of them - https://github.com/ar-io/ar-io-sdk/blob/main/.github/workflows/release.yml#L20-L24